src/main/kotlin/com/exactpro/th2/codec/api/IPipelineCodecFactory.kt

```diff
@@ -16,6 +16,11 @@
 
 package com.exactpro.th2.codec.api
 
-interface IPipelineCodecSettingsTypeProvider {
-    val type: Class<out IPipelineCodecSettings>
+import java.io.InputStream
+
+interface IPipelineCodecFactory : AutoCloseable {
+    val settingsClass: Class<out IPipelineCodecSettings>
+    fun init(dictionary: InputStream, settings: IPipelineCodecSettings? = null) // make dictionary optional too?
```

Comment:
`make dictionary optional too?`
A special codec can use lib to decode and verify messages, but we can pass lib as a dictionary, and we want to use any dictionary in th2 codec. Let's stay dictionary as a required argument.

Reply:
```suggestion
    fun init(dictionary: InputStream)
```